### DISCUSSION TITLE
Add additional fields to [authp].[RoleToPermissions] & [authp].[UserToRoles] table.

### DISCUSSION THREAD

**Discussion by moizsyed-10p** [JonPSmith/AuthPermissions.AspNetCore#93] [General]
What is the way to add more fields to authp tables? 
1. ** IsActive** field to handle whether the role is active or not.
2.  ** Custom Column** that can hold location data/Time data etc etc.

**Comment by JonPSmith** [comment]
Hi @moizsyed-10p,

The AuthP uses a "one database, but many schemas". For instance, in Example 3 (Single level multi-tenant application) the database has three parts:

- The AuthP data, e.g. AuthUser, RoleToPermissions, etc.
- The Tenant data, in this app has a simple invoice example
- The ASP.NET Core's individual user accounts data (allowing users to log in)

So, to add extra data you can create a new DbContext and add your own classes / tables. You can migrate your new DbContext via the `StartupServiceMigrateAnyDbContext<YourExtraDbContext>` - see the end of [the AuthP setup code](https://github.com/JonPSmith/AuthPermissions.AspNetCore/blob/main/Example3.MvcWebApp.IndividualAccounts/Program.cs#L52-L75) for how this works. I also recommend you add a schema to your tables to stop name clashing - see [this code in the AuthP DbContext](https://github.com/JonPSmith/AuthPermissions.AspNetCore/blob/7315a58bc59fcbe3dcf0ca6c8111b47f757ea2df/AuthPermissions.BaseCode/DataLayer/EfCode/AuthPermissionsDbContext.cs#L73).

To link your classes to the AuthP's classes I suggest you use EF Core's  [View mapping](https://learn.microsoft.com/en-us/ef/core/modeling/entity-types?tabs=data-annotations#view-mapping). I show how to do this in [this section](https://www.thereformedprogrammer.net/evolving-modular-monoliths-3-passing-data-between-bounded-contexts/#example-1-exchanging-data-via-the-database) in one of my "Evolving modular monoliths" articles.

**Reply by moizsyed-10p** [reply]
Hey  @JonPSmith ,

Thanks for the detailed reply.
I am following Example 4, and to add extra data I am using a new DbContext but my question was whether is there any possibility to add columns to the AuthP tables such as **AuthUser**, and **UserToRole**.

**Reply by JonPSmith** [reply]
No, you can't add new columns to the AuthP data - that's why I explained how you can create your DbContext and link to the the AuthP data.

The AuthP design uses a [Domain Driven Design](https://en.wikipedia.org/wiki/Domain-driven_design) structure.
